src/VSCodeExtension/src/quickPickJob.ts

```diff
@@ -0,0 +1,255 @@
+            if((!updateWorkspace && inputBox.step===4)||(updateWorkspace && inputBox.step===7)){
+                programArguments = inputBox.value;
+                inputBox.dispose();
+                quickPick.dispose();
+                resolve({jobName:jobName,provider:provider, target:target,programArguments:programArguments});
+            }
+          // user input job name so prompt program arguments input
+            if ((!updateWorkspace && inputBox.step===3)||(updateWorkspace && inputBox.step===6)){
+            jobName = inputBox.value;
+            setupProgramArguments(inputBox);
+        }
+        });
+
+         quickPick.onDidAccept(async()=>{
+            const selection = quickPick.selectedItems[0];
+            if (!selection || !selection["label"]) {
+                inputBox.dispose();
+                quickPick.dispose();
+                reject();
+              }
+              // user input target so prompt job name input
+              if ((!updateWorkspace && quickPick.step===2)||(updateWorkspace && quickPick.step===5)){
+                target = selection["label"];
+                quickPick.hide();
+                setupNameInputBox(inputBox);
+            }
+              // user input provider so prompt target input
+            if ((!updateWorkspace && quickPick.step===1)||(updateWorkspace && quickPick.step===4)){
+                provider = selection["label"];
+                setupTargetsQuickPick(quickPick,providersAndTargets,provider);
+            }
+
+         });
+         inputBox.onDidTriggerButton(async (button) => {
+            // user pressed name back button so prompt target input
+            if ((!updateWorkspace && inputBox.step===3)||(updateWorkspace && inputBox.step===6)) {
+                inputBox.hide();
+                setupTargetsQuickPick(quickPick,providersAndTargets,provider);
+            }
+            // user pressed program arguments back button so prompt job name input
+            if ((!updateWorkspace && inputBox.step===4)||(updateWorkspace && inputBox.step===7)) {
+                setupNameInputBox(inputBox);
+            }
+          });
+
+
+        quickPick.onDidTriggerButton(async (button) => {
+            // user pressed target back button so prompt provider input
+            if ((!updateWorkspace && quickPick.step===2)||(updateWorkspace && quickPick.step===5)) {
+                providersAndTargets = await setupProviderQuickPick(quickPick, context, quantumJobClient, workspaceInfo, inputBox);
+                if(providersAndTargets===undefined){
+                    inputBox.dispose();
+                    quickPick.dispose();
+                    reject();
+                }
+            }
+          });
+        quickPick.onDidHide(()=>{
+            quickPickIsHidden = true;
+
+        });
+
+
+
+     });
+
+
+}
+
+function setupNameInputBox(inputBox:vscode.InputBox){
+
+    inputBox.step = updateWorkspace?6:3;
```

Comment:
Same here. Step names instead of numbers will make this much easier to maintain.